### PR TITLE
Python3 offer tool feature

### PR DESCRIPTION
**Pull request by marekr**
This provides the feature for the python3 port to build the executable and make a working python install inside /tools.

Why?
There is a build process I have that requires the python executable to match the python library version being built against, I also need the python modules that don't exist in the python-embed packages that vcpkg_acquire_program retrieves.

This only adds the tool for Windows.

Now I expect problems and have questions for the vcpkg maintainers....

1. Python3 has important modules that used external libraries.

Currently it wants these exact versions of libraries which are tracked by in source script get_externals.bat
```
bzip2-1.0.6
libffi-3.3.0-rc0-r1
openssl-1.1.1f
 sqlite-3.31.1.0
tcl-core-8.6.9.0
tk-8.6.9.0
tix-8.4.3.6
xz-5.2.2 
zlib-1.2.11
```

of which it builds directly with the module source (not static or shared) and others it links shared. 

While I understand vcpkg policy is to use the split out ports. I do not think it's the best plan for python which calls out ~explicit~ versions which the vcpkg port system cannot respect as far I know.
 These shared libraries only get placed in the /share/python3/DLLs folder next to the /share/python3/Lib folder and not /bin which is the standard placement for these shared modules on windows python.

2. I made it install the core python modules in both /share/python3/DLLs and /tools/python3/DLLs. python.exe could be made to use /share/python3/DLLs in theory manually by users invoke it and setting to /share/python3

### CODE REVIEW

File: ports/python3/CONTROL [hunk 1]
```diff
@@ -7,3 +7,6 @@ Build-Depends: libffi, openssl, zlib (!uwp&!windows)
 
 Feature: enable-shared
 Description: Build shared libraries in addition to static ones built by default
+
```
Comment (NancyLi1013) [inline ports/python3/CONTROL:10]: Could you please update the `Port-Version`?
`Port-Version: 2`

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by Hoikas** [comment]
Duplicates part of #13556.

**Comment by marekr** [comment]
Defering to the other PR that's better and hopefully gets merged to make the python3 port usable ;)

**Comment by Hoikas** [comment]
I'm glad to see that I wasn't the only one frustrated by some of the aspects of the python3 port :smile:

**Pull request closed without merging**